Lsx branch

limshixun opened the pull request:
Added fully functional Tic-Tac-Toe with score board and reply function

limshixun committed:
Initial Commit

limshixun committed:
simple functionalities

limshixun committed:
add winning scenarios

limshixun committed:
add replay modal

limshixun committed:
add reset button

limshixun committed:
add score and scoreboard

sinwailam193 commented on src/app/components/Board.js:
```diff
@@ -0,0 +1,30 @@
+import './Board.css'
+
+function Board({handleBoxClick, board}) {
+
```
nit: line 6-8 can be deleted since it is just white space. If you're using visual studio, can install eslint to automatically fix your code in code editor

sinwailam193 replied:
also it seems like box.css file is not used, can remove it as well

sinwailam193 commented on src/app/page.js:
```diff
@@ -1,95 +1,116 @@
+const TicTacToe = () => {
+  const [board, setBoard] = useState(Array(9).fill(null));
+  const [xTurn, setXTurn] = useState(true);
+  const [status, setStatus] = useState("Next player: " + (xTurn ? "X" : "O"));
```
no need to include the entire string with the prefix of `Next player:` can just set X or O. Preferably just use a default, don't use a ternary operator within the default value of the useState

sinwailam193 commented on src/app/page.js:
```diff
@@ -1,95 +1,116 @@
+    }
+  };
+
+  function calculateWinner() {
```
`calculateWinner` this method can be moved out of the component and into a util function of its own in a separate file

sinwailam193 commented on src/app/styles.css:
```diff
@@ -0,0 +1,101 @@
+}
+
+.replay-button {
+margin-top: 20px;
```
indent the css lines so that it shows an indication that these css are within the brackets of `replay-button` and likewise for the other css

sinwailam193 reviewed (changes requested):
Overall is good, I recommend couple of things in this PR:
- Install a eslint tool to automatically beautify and edit your code to be aligned
- In the description of the PR, provide a detail explanation of what you did in the PR and if is UI changes, add an Image or even a video tutorial showcasing the features that you did.
- This PR is ok, but in the future break the PRs down into smaller pieces by features. For example, for the business logic of the game, can going into its own pr. The UI of the game, can go to another PR, the styling can also be its own pr. The objective is to break the PRs into smaller pieces so is easier for your reviewer to review.

limshixun committed:
Removes unnecessary files and lines and moves checkWinner function into util.js

limshixun committed:
add indent

limshixun commented:
1. Removed unnecessary empty lines and files
2. Fixed the format using eslint
3. Added indentation to CSS
4. Added new util.js for checkWinner function
